Sources/Commands/SwiftTestTool.swift

```diff
@@ -60,20 +86,36 @@ public class TestToolOptions: ToolOptions {
     var config: Build.Configuration = .debug
 
     /// If tests should run in parallel mode.
-    var shouldRunInParallel = false
+    var shouldRunParallel = false
 
     /// List the tests and exit.
     var shouldListTests = false
 
-    /// Run only these specified tests.
-    var specifier: String?
+    /// Run only tests matching this pattern
+    var filterPattern: String? = nil
+
+    /// Run only specific test
+    var specifier: String? = nil
+}
+
+/// Tests filtering specifier
+///
+/// This is used to filter tests to run
+///   .none     => No filtering
+///   .specific => Specify test with fully quantified name
+///   .regex    => RegEx pattern
+public enum TestCaseSpecifier {
+    case none
+    case specific(String)
+    case regex(String)
 }
 
 public enum TestMode {
     case version
-    case listTests
-    case run(String?)
-    case runInParallel
+    case listTests(TestCaseSpecifier)
+    case runSpecific(TestCaseSpecifier)
+    case runSerial(TestCaseSpecifier)
+    case runParallel(TestCaseSpecifier)
```

Comment:
I think instead of having the `TestCaseSpecifier` part of each enum, we can just add it to `TestToolOptions`.